src/physics/base_physics.cpp

```diff
@@ -40,17 +39,22 @@ void BasePhysics::setState(const std::vector<serac::GeneralCoefficient>& state_c
   SLIC_ASSERT_MSG(state_coef.size() == state_.size(), "State and coefficient bundles not the same size.");
 
   for (unsigned int i = 0; i < state_coef.size(); ++i) {
-    state_[i]->project(state_coef[i]);
+    state_[i].get().project(state_coef[i]);
   }
 }
 
-void BasePhysics::setState(const std::vector<std::shared_ptr<serac::FiniteElementState> >& state)
+void BasePhysics::setState(std::vector<serac::FiniteElementState>&& state)
 {
   SLIC_ASSERT_MSG(state.size() > 0, "State vector array of size 0.");
-  state_ = state;
+  // To avoid making a shallow copy of the references, we move individually
+  for (std::size_t i = 0; i < state.size(); i++) {
+    // Assigning to the FiniteElementState being referenced,
+    // not the reference itself
+    state_[i].get() = std::move(state[i]);
```

Comment:
Could you explain a bit why this is `std::move` instead of just assignment?

Reply:
This function isn't currently used anywhere, but I had envisioned it being used in a context where the vector of FEState objects would be constructed by the user only for passing to this function, which is why I required that the vector be moved into this function.  Since the vector is an rvalue, we can use move assignment on its elements instead of copy assignment, which isn't possible for `FiniteElementState` as things like a pointer to `FiniteElementCollection` can't be copied.

Let me know if I've misunderstood the intent of this function though - without seeing a use of this I wasn't 100% sure.

Reply:
I think I added this just because I thought it would be useful to have for building state from stuff we don't control. I think this assumption is fine as long as it's noted in the doxygen comments that this steals the data.